MAPL_Base/MAPL_EtaHybridVerticalCoordinate.F90

```diff
@@ -0,0 +1,282 @@
+#include "MAPL_Generic.h"
+#include "MAPL_ErrLog.h"
+#include "unused_dummy.H"
+
+module MAPL_EtaHybridVerticalCoordinate
+   use, intrinsic :: ISO_FORTRAN_ENV, only: REAL64, REAL32
+   use ESMF
+   use MAPL_ErrorHandlingMod
+   use MAPL_KeywordEnforcerMod
+   implicit none
+   private
+
+   public :: EtaHybridVerticalCoordinate
+
+
+   type :: EtaHybridVerticalCoordinate
+      private
+      real(kind=REAL64), allocatable :: ak(:)
+      real(kind=REAL64), allocatable :: bk(:)
+      integer :: num_levels = 0
+      real(kind=REAL64) :: ref_pressure
+   contains
+      procedure :: get_eta_r8
+      procedure :: get_eta_r4
+      procedure :: get_pressure_levels_r8
+      procedure :: get_pressure_levels_r4
+      procedure :: get_pressures_r8_3d
+      procedure :: get_pressures_r4_3d
+      generic :: get_eta =>get_eta_r8, get_eta_r4

```

Comment:
Add a module level interface that creates the object and returns ak, bk all in one swoop.

```f90
interface get_eta
   module procedure get_eta_onestep_r32
   module procedure get_eta_onestep_r64
end interface
```

```f90
subroutine get_eta_onstep_r32(filename, ptop, pint, ak, bk, unused, rc)
    ...
     type (EtaHybridVerticalCoordinate) :: vgrid
     vgrid = EtaHybridVerticalCoordinate(filename)
    call vgrid%get_eta(...)
end subroutine
```

Reply:
done. but 1) do wee need "get_pressure" interface? 2) do we need to make EtaHybridVerticalCoordinate private?